### PR TITLE
An ATL can view all expected results for a C4 Tests

### PR DESCRIPTION
**Pull request by ljades**
A new link on a product test, "View Expected Result Patients", allows an ATL to view expected results for C4 Tests with the same depth as the "View Patients" link.

Totals for populations also now appear at the footer of a table for a measure and test.

**Submitter:**
- [x] This pull request describes why these changes were made.
- [x] Internal ticket for this PR: https://jira.mitre.org/browse/CYPRESS-110
- [x] Internal ticket links to this PR
- [x] Code diff has been done and been reviewed
- [x] Tests are included and test edge cases
- [x] Tests have been run locally and pass

**Reviewer 1:**

Name: Robert Clark
- [ ] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [ ] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

**Reviewer 2:**

Name: Lauren DiCristofaro
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

### CODE REVIEW

File: app/views/records/by_filter_task.html.erb [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+<% cache [@records, @measure, hide_patient_calculation?] do %>
+  <div class="row">
+    <div id="records_list" class="col-sm-12">
+      <%= render partial: 'records_list', locals: { records: @records } %>
+    </div>
+  </div>
+<% end #cache records %>
+
+<script type="text/javascript">
```
Comment (rbclark) [inline app/views/records/by_filter_task.html.erb]: This seems to be a copy paste from `app/views/records/index.html.erb`, would it be possible to move this out into a proper JS file so this same javascript code can be reused?

Comment (rbclark) [reply app/views/records/by_filter_task.html.erb]: Actually it looks like there is no `#search_measures` block on this page at all, would it be possible to just remove the whole javascript block?

File: app/views/records/by_filter_task.html.erb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  <% end %>
+  <%= button_to good_results_task_path(@task), :method => :get, :class => "btn btn-default" do %>
+    <i class="fa fa-fw fa-download" aria-hidden="true"></i> Download HTML Patients
+  <% end %>
+</div>
+
+<% cache [@records, @measure, hide_patient_calculation?] do %>
+  <div class="row">
+    <div id="records_list" class="col-sm-12">
+      <%= render partial: 'records_list', locals: { records: @records } %>
```
Comment (rbclark) [inline app/views/records/by_filter_task.html.erb]: I am now getting `undefined local variable or method `measure' for #<#<Class:0x00007f806c7acf30>:0x00007f8064de4fe0>
Did you mean?  @measure` from this call when I navigate to by_filter_task.

**Comment by rbclark** [comment]
It may be good to have some cucumber tests which at least navigate to the by_filter_task page in order to catch errors like the one I was experiencing before in this code in the future.

**Comment by lmd59** [comment]
Should vendors be allowed to see expected patient results or just ATL/Admin roles?

**Comment by okeefm** [comment]
Just ATL/Admin, when we're not ignoring roles.

**Comment by okeefm** [comment]
@ljades I would agree with @rbclark here that we should write some Cucumber tests for this.

**Comment by dczulada** [comment]
Lets change the name from "View Expected Result Patients" to "View Expected Result"

Remove the original "Expected Patient Records (1)" section (on the Category I page)

screen shot 2017-12-11 at 10 43 14 am
